app/User.php

```diff
@@ -75,4 +76,23 @@ public function following()
     {
         return $this->belongsToMany('App\User', 'relationships', 'follower_id', 'following_id')->withTimestamps();
     }
+
+    /**
+     * Update User Profile
+     *
+     * @param object $data
+     * @return void
+     */
+    public function updateProfile($data)
+    {
+        $this->update([
+            'first_name'  => $data->first_name ?: $this->first_name,
+            'last_name'   => $data->last_name ?: $this->last_name,
+            'email'       => $data->email ?: $this->email,
+            'password'    => $data->new_password ? Hash::make($data->new_password) : $this->password,
+            'avatar_path' => $data->hasFile('avatar') && $data->file('avatar')->isValid()
+                ? 'data:' . $data->file('avatar')->getClientMimeType() . ';base64,' . base64_encode(file_get_contents($data->file('avatar')->getRealPath()))
```

Comment:
I suggest store this one in variable in following reason:

* To define what is this data
* To keep at least 80 characters per line as mentioned in the [PSR-2](https://www.php-fig.org/psr/psr-2/#1-overview) (Coding Standards)

Reply:
I defined this data and changed code more readable.